### PR TITLE
vagrant: 2.4.1 -> 2.4.3

### LINKED ISSUE #354352: Update request: Vagrant 2.4.1 → 2.4.3
- Package name: Vagrant
- Latest released version: [2.4.3](https://github.com/hashicorp/vagrant/releases/tag/v2.4.3)
- Current version on the unstable channel: 2.4.1
- Current version on the stable/release channel: 2.4.1
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@tylerjl 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by tylerjl**
This has been marked broken on Darwin and Linux; after updating to latest stable and bumping dependencies in the vagrant-libvirt gemset (there were conflicting/outdated dependency conflicts after generating their Gemfile locks) I can vouch that it works on x86_64-linux at the very least (both building and managing a libvirt box - I tested w/RDP as well)

Resolves #354352 #348938

I have not built this on darwin, so I left the `broken` stanza in-place for that platform. If someone could checkout my branch and report success || failure on Darwin I'd appreciate it (then I'll remove `meta.broken`).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by bjornfor — COMMENTED** [review]
Perfect timing! I've spent over a day trying to upgrade this package. I kept getting conflicts and hash mismatches. (I did `bundler lock && bundix` in vagrant and vagrant-libvirt source trees, but hit various build issues after that.)

Can you describe in the commit message exactly what you did?

File: pkgs/by-name/va/vagrant/package.nix [hunk 1]
```diff
@@ -118,6 +118,6 @@ in buildRubyGem rec {
     license = licenses.bsl11;
     maintainers = with maintainers; [ tylerjl ];
     platforms = with platforms; linux ++ darwin;
-    broken = true; # build fails on darwin and linux
+    broken = stdenv.hostPlatform.isDarwin;
```
Comment (andersk) [inline pkgs/by-name/va/vagrant/package.nix:121]: I tested that this builds and runs on Darwin, so we can remove this.

Comment (tylerjl) [reply pkgs/by-name/va/vagrant/package.nix:121]: Thank you! I'll remove the `broken` stanza.

**Comment by tylerjl** [comment]
> Perfect timing! I've spent over a day trying to upgrade this package. I kept getting conflicts and hash mismatches. (I did `bundler lock && bundix` in vagrant and vagrant-libvirt source trees, but hit various build issues after that.)
> 
> Can you describe in the commit message exactly what you did?

Sure thing; I'm going to remove `broken` so I'll rebase with an expanded commit message.

Long term I (or somebody) needs to punch out a small script to help generate updates because it.. isn't straightforward. The tl;dr is

- Clone the vagrant repository; check out the desired version tag
- `bundle lock` then `bundix`; overwrite `gemset.nix` 
- When crawling the diff I noted the old gemset doesn't include the `vagrant` gem itself. Unsure if this breaks the build or not but I removed it to stay consistent
- Clone vagrant-libvirt, check out the desired tag
- I _commented out_ the `development` and `test` groups before `bundle lock` because they pulled in problematic dependencies that clashed with `vagrant` and `bundle` doesn't accept a group argument for `lock`. 
- `bundix` and tweak `vagrant-libvirt` in `gemset.nix` to change the source URL from a relative path
- Copy over the vagrant-libvirt `gemset.nix`

Unfortunately this seems like a very brittle process since both upstream projects don't ship their own `Gemfile.lock` files and a `bundle lock`  is inherently time-dependent and there isn't much of a guarantee you'd get the same result a month from now, but at least we have a functional commit at this point in time.

**Review by bjornfor — APPROVED** [review]
Thanks!

**Comment by bjornfor** [comment]
Now I wonder whether it's a good idea to write an update script, or if we should just have the upgrade steps as a comment in .../vagrant/package.nix?

**Comment by andersk** [comment]
An update script would be great, but we should merge this first to solve the build failure.

**Comment by bjornfor** [comment]
> An update script would be great, but we should merge this first to solve the build failure.

Definitely!

**Pull request merged**